### PR TITLE
Disable tests that use databox when spiner is off

### PR DESCRIPTION
**Pull request by mauneyc-LANL**
## PR Summary

@jhp-lanl hit an issue where `SINGULARITY_USE_SPINER=OFF` would fail tests that used `spiner`. In a way, the tests worked!

This disables building those tests when `SINGULARITY_USE_SPINER=OFF`

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by using the `make format` command after configuring with `cmake`.
- [x] Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.
- [x] After creating a pull request, note it in the CHANGELOG.md file
- [ ] If preparing for a new release, update the version in cmake.

### CODE REVIEW

**Review by jhp-lanl — APPROVED** [review]
I think it will take significant effort to make these tests run without databoxes. This seems like an appropriate near-term solution to me

**Review by Yurlungur — APPROVED** [review]

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -40,7 +40,7 @@ if(SINGULARITY_USE_EOSPAC AND SINGULARITY_TEST_SESAME)
                         PRIVATE Catch2::Catch2 singularity-eos::singularity-eos)
 endif()
 
-if(SINGULARITY_BUILD_CLOSURE)
+if(SINGULARITY_BUILD_CLOSURE AND SINGULARITY_USE_SPINER)
```
Comment (dholladay00) [inline test/CMakeLists.txt:43]: I think we should add fortran to the if list.

**Comment by dholladay00** [comment]
> I think it will take significant effort to make these tests run without databoxes. This seems like an appropriate near-term solution to me

I'm not sure that I agree. `portablemdarray` could be dropped in w/o much additional effort I think.

File: test/CMakeLists.txt (hunk 1 above)
Comment (mauneyc-LANL) [reply test/CMakeLists.txt:43]: The closure code is built independent of Fortran, `singularity-eos/CMakeLists.txt`

```cmake
if (SINGULARITY_BUILD_CLOSURE)
  list(APPEND EOS_SRCS eos/singularity_eos.cpp)
  list(APPEND EOS_HEADERS eos/singularity_eos.hpp closure/mixed_cell_models.hpp)
endif()

if (SINGULARITY_USE_FORTRAN)
  list(APPEND EOS_SRCS eos/singularity_eos.f90)
  # would rather handle this more robustly, being sloppy for now
  list(APPEND EOS_MODS singularity_eos.mod singularity_eos_types.mod)
endif()
```

**Comment by jhp-lanl** [comment]
> > I think it will take significant effort to make these tests run without databoxes. This seems like an appropriate near-term solution to me
> 
> I'm not sure that I agree. `portablemdarray` could be dropped in w/o much additional effort I think.

That's good to hear then! I took a brief look at it and didn't know the code enough to make that assessment. I would propose making an issue to do that and move forward with this since it correctly aligns the code with its dependencies.

File: test/CMakeLists.txt (hunk 1 above)
Comment (dholladay00) [reply test/CMakeLists.txt:43]: Yes, but you'd be saving me some time when I have to remember to do this in my PR b/c soon it'll only be built when also fortran.

**Comment by Yurlungur** [comment]
> That's good to hear then! I took a brief look at it and didn't know the code enough to make that assessment. I would propose making an issue to do that and move forward with this since it correctly aligns the code with its dependencies.

Issue #289 . As I mention there and in mattermost, we may want to investigate moving some of the memory management niceties out of spiner and into ports of call. That might be good for the project as a whole anyway.

**Comment by jhp-lanl** [comment]
> > That's good to hear then! I took a brief look at it and didn't know the code enough to make that assessment. I would propose making an issue to do that and move forward with this since it correctly aligns the code with its dependencies.
> 
> Issue #289 . As I mention there and in mattermost, we may want to investigate moving some of the memory management niceties out of spiner and into ports of call. That might be good for the project as a whole anyway.

I agree. I think that will be useful for other projects that depend on `ports_of_call`

File: test/CMakeLists.txt (hunk 1 above)
Comment (dholladay00) [reply test/CMakeLists.txt:43]: After looking at it closer, you're correct, this test can remain but other things will need to be ifdef'd out.

Comment (dholladay00) [reply test/CMakeLists.txt:43]: Requested change rescinded.

**Comment by dholladay00** [comment]
@Yurlungur, @jhp-lanl while all of that may be true. I am sensing scope creep here. I think the current tests could be implemented with portablemdarray as-is and that should be the focus for now.

**Comment by jhp-lanl** [comment]
> @Yurlungur, @jhp-lanl while all of that may be true. I am sensing scope creep here. I think the current tests could be implemented with portablemdarray as-is and that should be the focus for now.

I'm not familiar enough with the MDArrays to be honest. But I still think this MR should go through in the meantime

**Comment by Yurlungur** [comment]
> while all of that may be true. I am sensing scope creep here. I think the current tests could be implemented with portablemdarray as-is and that should be the focus for now.

I think we should just move forward with disabling these tests when spiner is disabled for now. And we can rethink/refactor later.

**Pull request merged**